sdk/core/scheduler/thread.h

```diff
@@ -564,8 +571,12 @@ namespace
 		///@}
 		/// Pointer to the list of the resource this thread is blocked on.
 		ThreadImpl **sleepQueue;
-		/// If suspended, when will this thread expire. The maximum value is
-		/// special-cased to mean blocked indefinitely.
+		/**
+		 * If suspended, when will this thread expire. The maximum value is
+		 * special-cased to mean blocked indefinitely.
+		 *
+		 * When a thread is running, this the time at which it was scheduled.
```

Comment:
Again, just to check my understanding: Time of the _last_ scheduling event and not the time of any _upcoming_ scheduling event because there might not actually be one, if this is the only runnable thread in the system, and besides it's cheaper to not do the addition unless someone asks in `has_run...`?

Reply:
Yes, this when you were scheduled.  When you do futex wake, you may have been running for more then one tick, or less than one tick.